synapse/events/spamcheck.py

```diff
@@ -214,3 +216,48 @@ async def check_registration_for_spam(
                     return behaviour
 
         return RegistrationBehaviour.ALLOW
+
+    async def check_media_file_for_spam(
+        self, file_wrapper: ReadableFileWrapper, file_info: FileInfo
+    ) -> bool:
+        """Checks if a piece of newly uploaded media should be blocked.
+
+        This will be called for local uploads, downloads of remote media, each
+        thumbnail generated for those, and web pages/images used for URL
+        previews.
```

Comment:
It seems a bit odd to me to do this for thumbnail generation, but I think it just due to where we're hooking into the code here?

Reply:
Yeah. I can make it not do thumbnails, but its not clear that is the right thing to do. Spam checkers can check that themselves.